master/buildbot/status/web/builder.py

```diff
@@ -191,29 +191,32 @@ def performAction(self, req):
         # send the user back to the builder page
         defer.returnValue(path_to_return)
 
+def buildForcePropertyName(scheduler, field):
+    return "%s.%s"%(scheduler.name, field.fullName)
+
 def buildForceContextForField(req, default_props, sch, field, master, buildername):
-    pname = "%s.%s"%(sch.name, field.fullName)
-    
-    default = field.default
-    
+    pname = buildForcePropertyName(sch, field)
+    default = req.args.get(pname, [field.default])[0]
+
+    buildForceContextForSingleFieldWithValue(default_props, sch, field, master, buildername, default)
+
+    if "nested" in field.type:
+        for subfield in field.fields:
+            buildForceContextForField(req, default_props, sch, subfield, master, buildername)
+
+def buildForceContextForSingleFieldWithValue(default_props, scheduler, field, master, builderName, value):
+    pname = buildForcePropertyName(scheduler, field)
+
     if "list" in field.type:
-        choices = field.getChoices(master, sch, buildername)
+        choices = field.getChoices(master, scheduler, builderName)
         if choices:
-            default = choices[0]
+            value = next((x for x in choices if x == value), choices[0])
```

Comment:
I really like the way you've created this generator and used next but it's a bit over-engineeredl. Entire existence checking could be simplified to following lines
```
if choices and value not in choices:
    value = choices[0]
```

What if choices is an empty list and what will be assigned in line 219?

Reply:
:+1: